importer/helper.go

```diff
@@ -237,27 +238,48 @@ func loadGenesis(db idb.IndexerDb, in io.Reader) (err error) {
 }
 
 // InitialImport imports the genesis block if needed. Returns true if the initial import occurred.
-func InitialImport(db idb.IndexerDb, genesisJSONPath string, l *log.Logger) bool {
-	state, err := db.GetImportState()
+func InitialImport(db idb.IndexerDb, genesisJSONPath string, client *algod.Client, l *log.Logger) bool {
+	log.Infof("Initial import. Client is null:  %t", client == nil)
+	_, err := db.GetImportState()
 
-	// Only import when the state is not found.
-	if err == idb.ErrorNotInitialized {
-		state.AccountRound = -1
-		if genesisJSONPath != "" {
-			l.Infof("loading genesis %s", genesisJSONPath)
-			// if we're given no previous state and we're given a genesis file, import it as initial account state
-			gf, err := os.Open(genesisJSONPath)
-			maybeFail(err, l, "%s: %v", genesisJSONPath, err)
-			err = loadGenesis(db, gf)
-			maybeFail(err, l, "%s: could not load genesis json, %v", genesisJSONPath, err)
-			return true
+	// Exit immediately or crash if we don't see ErrorNotInitialized.
+	if err != idb.ErrorNotInitialized {
+		maybeFail(err, l, "getting import state, %v", err)
+		return false
+	}
+
+	// Import genesis file from file or algod.
+	//var genesisString string
+	var genesisReader io.Reader
+
+	// Attempt to read file.
+	if genesisJSONPath != "" {
+		l.Infof("loading genesis file %s", genesisJSONPath)
+		genesisReader, err = os.Open(genesisJSONPath)
+		if err != nil {
+			genesisReader = nil
```

Comment:
I think if file path is specified but we can't read from it, we should return an error.